src/main/java/parkinglot/ParkingLot.java

```diff
@@ -0,0 +1,131 @@
+package parkinglot;
+
+public class ParkingLot {
+
+    private String name;
+    private String address;
+    private int capacity;
+
+    private int freeSpots = 0;
+
+    private ParkingSpot[] parkingSpots;
+    private ParkingTicket[] parkingTickets;
+    private Vehicle[] vehicles;
+
+
+    // initiate Parking Lot
+    ParkingLot(int capacity) {
+
+        //System.out.println("Initialization of new parking lot with capacity: " + capacity);
+        this.capacity = capacity;
+
+        parkingSpots = new ParkingSpot[capacity];
+        parkingTickets = new ParkingTicket[capacity];
+        vehicles = new Vehicle[capacity];
+
+        for (int i = 0; i < capacity; i++) {
+            parkingSpots[i] = new ParkingSpot(i + 1);
+            freeSpots++;
+        }
+    }
+
+    private boolean isFull () {
+        if (freeSpots == 0) {
+            return true;
+        }
+        return false;
+    }
+
+    public String addVehicleToParkingLot(Vehicle vehicle) {
+
+        if (isFull()) {
+            return "Sorry, but parking spot is full. Try again later.";
+        } else {
+            freeSpots--;
+
+            int freeParkingSpot = checkFirstFreeParkingSpot();
+            parkingSpots[freeParkingSpot].addVehicle(vehicle);
+
+            parkingTickets[freeParkingSpot] = new ParkingTicket();
+
+            this.vehicles[freeParkingSpot] = vehicle;
+
+            return  display("Add", freeParkingSpot);
+        }
+    }
+
+    public String removeVehicleFromParkingLot(Vehicle vehicle, String parkingTicketNumber) {
+        freeSpots++;
+        int spot = -1;
+
+        for (int i = 0; i < capacity; i++) {
+            if (parkingTickets[i].getParkingTicketNumber() == parkingTicketNumber) {
```

Comment:
you should compare strings using equals(), not ==

Reply:
I agree. Will be corrected in next task

Reply:
Corrected. https://github.com/AqaCoreSchool/homeworks/pull/28/commits/57099748843c75dfa4f31fcca940318807f1aa55